### PR TITLE
[New] Add maptalks@0.37.0 w/ npm auto-update

### PR DESCRIPTION
**Pull request by finch185277**
Pull request for issue: #12432

Checklist for **Pull request** or **lib adding request issue** follows the conventions.

Note that if you are using a distribution purpose repository/package, please also provide the url and other related info like popularity of the source code repo/package.

# Profile of the lib
 * Git repository (required):https://github.com/maptalks/maptalks.js
 * Official website (optional, not the repository):https://maptalks.org
 * NPM package url (optional):https://www.npmjs.com/package/maptalks
 * License and its reference:
 * GitHub / Bitbucket popularity (required):
   - Count of watchers:81
   - Count of stars:1215
   - Count of forks:115
 * NPM download stats (optional):
   - Downloads in the last day:110
   - Downloads in the last week:265
   - Downloads in the last month:1531

# Essential checklist
 * [ ] I'm the author of this library
   * [ ] I would like to add link to the page of this library on CDNJS on website / readme
 * [x] This lib was not found on cdnjs repo
 * [x] No already exist / duplicated issue and PR
 * [x] The lib has notable popularity
   * [x] More than 200 [Stars / Watchers / Forks] on [GitHub / Bitbucket]
   * [x] More than 800 downloads stats per month on npm registry
 * [x] Project has public repository on famous online hosting platform (or been hosted on npm)

# Auto-update checklist
 * [x] Has valid tags for each versions (for git auto-update)
 * [x] Auto-update setup
 * [x] Auto-update target/source is valid.
 * [x] Auto-update filemap is correct.

# Git commit checklist
 * [x] The first line of commit message is less then 50 chars, be clean and clear, easy to understand.
 * [x] The parent of the commit(s) in the PR is not old than 3 days.
 * [x] Pull request is sending from a non-master branch with meaningful name.
 * [ ] Separate unrelated changes into different commits.
 * [x] Use rebase to squash/fixup dummy/unnecessary commits into only one commit.
 * [x] Close corresponding issue in commit message
 * [ ] Mention related issue(s), people in commit message, comment.

### CODE REVIEW

**Comment by finch185277** [comment]
Add images from GitHub.

File: ajax/libs/maptalks/package.json [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    "type": "git",
+    "url": "https://github.com/maptalks/maptalks.js.git"
+  },
+  "homepage": "https://maptalks.org",
+  "npmName": "maptalks",
+  "npmFileMap": [
+    {
+      "basePath": "dist",
+      "files": [
+        "**/*"
```
Comment (sashberd) [inline ajax/libs/maptalks/package.json]: In some very old versions there are Plugins folder and maptalks-demo.js files. I do not sure that we need them.  See v0.2.0. @sufuf3 please help with this issue

Comment (sufuf3) [inline ajax/libs/maptalks/package.json]: Hey, @finch185277 did you remember how to use auto-updater?
After you finish the pattern, you can try `./auto-update.js run maptalks` to check everything grab the right files. (you can use the command `tree` & `du` ... etc to help you check file structure)

**This patten should be fixed.** Would you try to exclude unnecessary files which can't be used in web front-end by rewrite the pattern? Like `ajax/libs/maptalks/0.1.0/maptalks.min.js.gz`, `ajax/libs/maptalks/0.6.0/canvg`, `ajax/libs/maptalks/0.7.0/api/*.html`, `ajax/libs/maptalks/0.26.3/images/control/Thumbs.db`, `ajax/libs/maptalks/0.30.2/LICENSE`, `ajax/libs/maptalks/0.30.2/ACKNOWLEDGEMENT` ... etc

Also, if you find some file name include a version number, you have to try to manually add them.
Besides, can `maptalks.es.js` be used in web front-end?

If you don't know which file needs to be hosted, you could open an issue on upstream's git repo to discuss the author with the problem you met.

Comment (finch185277) [reply ajax/libs/maptalks/package.json]: I can use 
```
      "files": [
        "**/!(ACKNOWLEDGEMENT|LICENSE|*.es.js|*.gz)"
      ]
```
to shrink the scope.
But how to restrict `images/control/Thumbs.db`, `canvg/*`, `api/*.html`?
@sufuf3  Would you like to show me some examples?

**Pull request merged**